build.gradle.kts

```diff
@@ -0,0 +1,46 @@
+/*
+   Copyright 2017-2021 Charles Korn.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+val gradleWrapperVersion: String by project
+
+plugins {
+    id("com.diffplug.spotless")
+    id("com.github.ben-manes.versions")
+    id("org.ajoberstar.reckon")
+    kotlin("jvm") apply false
+    kotlin("plugin.serialization") apply false
+}
+
+apply(from = "$rootDir/gradle/spotless.gradle")
+apply(from = "$rootDir/gradle/updateInfoUpload.gradle.kts")
+apply(from = "$rootDir/gradle/utilities.gradle")
+apply(from = "$rootDir/gradle/versioning.gradle")
+
+tasks {
+    register<Copy>("assembleRelease") {
+        description = "Prepares files for release."
+        group = "Distribution"
+
+        from(project("app").getTasksByName("shadowJar", false))
+        from(project("wrapper").getTasksByName("build", false))
+        into(getBuildDir().toPath().resolve("release"))
+    }
+
+    withType<Wrapper> {
+        gradleVersion = gradleWrapperVersion
```

Comment:
Is this required? (It wasn't in the old `build.gradle`.)

Reply:
Not required.  It must have been on my personal fork.  Should I remove?

Reply:
It ensures that you can edit `gradle.properties` to update Gradle version, and then run `./gradlew wrapper` to bump the shell scripts.

Reply:
I'd prefer to remove it - otherwise I don't think Renovate will be able to update it automatically ([the docs](https://docs.renovatebot.com/modules/manager/gradle-wrapper/) say it will only look in `gradle-wrapper.properties` by default, and I'm not sure how it would deal with the version being in two places).